src/network/network.cpp

```diff
@@ -857,7 +857,7 @@ static void NetworkInitGameInfo()
  */
 bool NetworkValidateServerName(std::string &server_name)
 {
-	StrTrimInPlace(server_name);
+	server_name = StrTrimView(server_name);
```

Comment:
Assigning a string a borrowed view of itself?
I see the old code did the same, but I don't think this is valid.

The other changes in this commit do the same thing.
I think we need to keep `StrTrimInPlace`, and make it use `std::string::erase` instead.